stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/ChannelListView.kt

```diff
@@ -158,8 +162,8 @@ public class ChannelListView @JvmOverloads constructor(
 
     public override fun onVisibilityChanged(view: View, visibility: Int) {
         super.onVisibilityChanged(view, visibility)
-        if (visibility == View.VISIBLE) {
-            requireAdapter().notifyDataSetChanged()
+        if (visibility == View.VISIBLE && ::adapter.isInitialized) {
+            adapter.notifyDataSetChanged()
```

Comment:
`requireAdapter()` initializes the adapter when it is not initialized yet and we cannot set custom view holder if the adapter is already initialized.

Reply:
I guess this implementation changed recently. The way I had it configured previously, you could set a VHF at any point.

Reply:
Yes, sorry, this was me in https://github.com/GetStream/stream-chat-android/pull/1163, where I change the channel list API to match how the message list API works